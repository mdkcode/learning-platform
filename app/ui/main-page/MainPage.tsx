"use client";
import Stats from "@/app/ui/main-page/Stats/Stats";
import ReadingImg from "assets/img/reading.svg";
import Button from "@/app/ui/components/Button/Button";
import { ButtonType } from "../components/Button/utils";
import { handleSignIn } from "@/app/api/auth/[...nextauth]/auth";

const MainPage = () => {
  return (
    <div className="lg:h-screen flex flex-col lg:flex-row items-center justify-between">
      <div className="max-w-[500px] xl:max-w-[650px]">
        <h1 className="text-[#00004B] text-5xl xl:text-7xl font-extrabold leading-[4rem]">
          Maximize skill, minimize budget
        </h1>
        <p className="text-lg text-gray-500 my-7">
          Our modern courses across a range of in-demand skills will give you
          the knowledge you need the life you want.
        </p>
        <Button
          label="Get started"
          buttonType={ButtonType.NAVIGATION}
          className="bg-gradient-to-r from-orange-400 to-pink-500 hover:bg-gradient-to-r hover:from-orange-500 hover:to-pink-600"
          onClick={() => handleSignIn()}
        />
      </div>
      <div className="relative">
        <Stats
          label="Members"
          value="29k"
          className="absolute right-10  top-0"
        />
        <div className="h-auto w-full max-w-[400px] lg:max-w-[700px]">
          <ReadingImg className="w-full h-auto" />
        </div>
        <Stats
          label="Courses"
          value="1,041"
          className="absolute left-10 top-12"
        />
      </div>
    </div>
  );
};

export default MainPage;
