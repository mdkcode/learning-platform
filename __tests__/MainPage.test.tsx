import { render } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import MainPage from "@/app/ui/main-page/MainPage";

vi.mock("assets/img/reading.svg", () => ({
  default: () => <svg data-testid="mock-reading-img" />,
}));

vi.mock("@/app/api/auth/[...nextauth]/auth", () => ({
  handleSignIn: vi.fn(),
}));

describe("MainPage", () => {
  it("matches snapshot", () => {
    const { container } = render(<MainPage />);
    expect(container).toMatchSnapshot();
  });
});
