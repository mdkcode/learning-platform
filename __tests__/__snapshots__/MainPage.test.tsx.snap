// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainPage > matches snapshot 1`] = `
<div>
  <div
    class="lg:h-screen flex flex-col lg:flex-row items-center justify-between"
  >
    <div
      class="max-w-[500px] xl:max-w-[650px]"
    >
      <h1
        class="text-[#00004B] text-5xl xl:text-7xl font-extrabold leading-[4rem]"
      >
        Maximize skill, minimize budget
      </h1>
      <p
        class="text-lg text-gray-500 my-7"
      >
        Our modern courses across a range of in-demand skills will give you the knowledge you need the life you want.
      </p>
      <button
        class="px-7 py-3 text-white font-semibold rounded-lg bg-gradient-to-r from-orange-400 to-pink-500 hover:bg-gradient-to-r hover:from-orange-500 hover:to-pink-600 bg-gradient-to-r from-orange-400 to-pink-500 hover:bg-gradient-to-r hover:from-orange-500 hover:to-pink-600"
      >
        Get started
      </button>
    </div>
    <div
      class="relative"
    >
      <div
        class="p-4 pr-8 rounded-lg bg-gray-100 inline-block absolute right-10  top-0"
      >
        <p
          class="font-bold text-gray-500"
        >
          Members
        </p>
        <h4
          class="text-3xl font-extrabold text-[#00004B]"
        >
          29k
        </h4>
      </div>
      <div
        class="h-auto w-full max-w-[400px] lg:max-w-[700px]"
      >
        <svg
          data-testid="mock-reading-img"
        />
      </div>
      <div
        class="p-4 pr-8 rounded-lg bg-gray-100 inline-block absolute left-10 top-12"
      >
        <p
          class="font-bold text-gray-500"
        >
          Courses
        </p>
        <h4
          class="text-3xl font-extrabold text-[#00004B]"
        >
          1,041
        </h4>
      </div>
    </div>
  </div>
</div>
`;
